pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -132,15 +140,22 @@ func (c *Controller) Apply(change *mcpclient.Change) error {
 			}
 		}
 
+		// adjust the type name for mesh-scoped resources
+		typ := descriptor.Type
+		if namespace == "" && descriptor.Type == model.AuthenticationPolicy.Type {
+			typ = model.AuthenticationMeshPolicy.Type
```

Comment:
Does this mean we always create config with type of `mesh-policy` for either `policy` and `mesh-policy`?

Reply:
No, it means that resources of type `istio.authentication.v1alpha1.Policy` are converted to `mesh-policy` if the namespace is omitted (i.e. cluster-scoped) and `policy` if the namespace is specified (i.e. namespace-scoped).